ford/graphs.py

```diff
@@ -418,6 +421,11 @@ def __init__(self, obj, gd, hist=None):
         # ToDo: Figure out appropriate way to handle interfaces to routines in submodules.
         self.proctype = getattr(obj, "proctype", "")
         super().__init__(obj, gd)
+
+        if gd.show_proc_parent:
+            if hasattr(obj, "parent"):
+                self.attribs["label"] = obj.parent.name + "::" + self.name
```

Comment:
`parent` can be `None`, so we actually want:

```suggestion
            if parent := getattr(obj, "parent", None):
                self.attribs["label"] = f"{parent.name}::{self.name}"
```

I've used the [walrus operator](https://realpython.com/python-walrus-operator/) to avoid more nested ifs, and an [f-string](https://realpython.com/python-f-strings/) here as well to avoid the string concatenation